File: Data Structures & Algorithms/count-odd-numbers-in-an-interval-range/submission-2.java
class Solution {
    public int countOdds(int low, int high) {
        int cnt = 0;

        while (low <= high) {
            if (low == high && low % 2 == 1 && high % 2 == 1) {
                cnt++;
                break;
            }

            if (low % 2 == 1) cnt++;

            if (high % 2 == 1) cnt++;

            low++;
            high--;
        }


        return cnt;
    }
}

File: Data Structures & Algorithms/partition-to-k-equal-sum-subsets/submission-2.py
class Solution:
    def canPartitionKSubsets(self, nums: List[int], k: int) -> bool:
        if sum(nums) % k != 0:
            return False
        
        N = len(nums)
        target = sum(nums) // k
        cnt = [0]

        def dfs(start, target):
            if target == 0:
                cnt[0] += 1
                return
            
            if target < 0:
                return

            for i in range(start, N):
                if nums[i] <= target:
                    target -= nums[i]
                    dfs(i + 1, target)
                    target += nums[i]
        
        dfs(0, target)
        return cnt[0] == k